### PR TITLE
Allow to configure listen address by env variable

### PR DESCRIPTION
**Pull request by okurz**
Related to https://github.com/os-autoinst/openQA/pull/3431

Related progress issue: https://progress.opensuse.org/issues/69355

### CODE REVIEW

**Review by Martchus — COMMENTED** [review]
I suppose the `_api` functions (e.g. `OpenQA::WebSockets::Client::_api`) need to be changed as well.

**Review by ilausuch — APPROVED** [review]
What is done until now, seems good for me

File: t/44-scripts.t [hunk 1]
```diff
@@ -16,7 +16,8 @@
 
 use Test::Most;
 
-use FindBin '$Bin';
+use FindBin;
+use lib "$FindBin::Bin/lib";
```
Comment (kalikiana) [inline t/44-scripts.t:20]: Interestingly this causes a compliation error even though we do exactly the same in `03-auth.t` 🤔

Comment (okurz) [reply t/44-scripts.t]: that part is already covered by a different PR, rebased and resolved.

**Review by kalikiana — APPROVED** [review]

File: lib/OpenQA/WebSockets/Client.pm [hunk 2]
```diff
@@ -61,8 +62,9 @@ sub singleton { state $client ||= __PACKAGE__->new }
 
 sub _api {
     my ($self, $method) = @_;
-    my $port = $self->port;
-    return "http://127.0.0.1:$port/api/$method";
+    my $address = $self->address;
+    my $port    = $self->port;
+    return "http://$address:$port/api/$method";
```
Comment (Martchus) [inline lib/OpenQA/WebSockets/Client.pm:67]: Looks good but you need to adjust `OpenQA::Scheduler::Client` in the same way.

**Review by Martchus — CHANGES_REQUESTED** [review]

File: lib/OpenQA/WebSockets/Client.pm (hunk 2 above)
Comment (kalikiana) [reply lib/OpenQA/WebSockets/Client.pm:67]: @okurz

**Comment by kalikiana** [comment]
Do we still want this? Considering both referenced tickets are resolved now 🤔

**Comment by okurz** [comment]
maybe eventually we want this again :)

**Pull request closed without merging**